packages/subgraph/src/controller.ts

```diff
@@ -70,213 +77,314 @@ import { loadOrCreateAccount, BIGINT_ONE, BIGINT_ZERO } from "./util"
-    hourStatSnapshot.totalCollateralAmount  = totalCollateral
-    hourStatSnapshot.timestamp = hourStartUnixBigInt
+  if (
+    hourStatSnapshot !== null &&
+    hourStatSnapshot.timestamp.notEqual(hourStartUnixBigInt)
+  ) {
+    let totalCollateral = hourStatSnapshot.totalCollateralAmount;
+    hourStatSnapshot.id = hourStatSnapshot.timestamp.toString();
+    hourStatSnapshot.save();
+
+    hourStatSnapshot = new HourStatSnapshot("last");
+    hourStatSnapshot.totalCollateralAmount = totalCollateral;
+    hourStatSnapshot.timestamp = hourStartUnixBigInt;
   }
 
-  return hourStatSnapshot as HourStatSnapshot
-
+  return hourStatSnapshot as HourStatSnapshot;
 }
 
 function getDayStatSnapshot(timestamp: BigInt): DayStatSnapshot {
+  let dayID = timestamp.toI32() / 86400;
+  let dayStartUnix = dayID * 86400;
+  let dayStartUnixBigInt = BigInt.fromI32(dayStartUnix);
 
+  let dayStatSnapshot = DayStatSnapshot.load("last");
 
-  let dayID = timestamp.toI32() / 86400
-  let dayStartUnix = dayID * 86400
-  let dayStartUnixBigInt = BigInt.fromI32(dayStartUnix)
-
-  let dayStatSnapshot = DayStatSnapshot.load('last')  
+  if (dayStatSnapshot === null) {
+    dayStatSnapshot = new DayStatSnapshot("last");
+    dayStatSnapshot.totalCollateralAmount = BIGINT_ZERO;
+    dayStatSnapshot.timestamp = dayStartUnixBigInt;
+  }
 
-  if (dayStatSnapshot === null) { 
-    dayStatSnapshot = new DayStatSnapshot('last')
-    dayStatSnapshot.totalCollateralAmount  = BIGINT_ZERO
-    dayStatSnapshot.timestamp = dayStartUnixBigInt
+  if (
+    dayStatSnapshot !== null &&
+    dayStatSnapshot.timestamp.notEqual(dayStartUnixBigInt)
+  ) {
+    let totalCollateral = dayStatSnapshot.totalCollateralAmount;
+    dayStatSnapshot.id = dayStatSnapshot.timestamp.toString();
+    dayStatSnapshot.save();
+
+    dayStatSnapshot = new DayStatSnapshot("last");
+    dayStatSnapshot.totalCollateralAmount = totalCollateral;
+    dayStatSnapshot.timestamp = dayStartUnixBigInt;
   }
 
+  return dayStatSnapshot as DayStatSnapshot;
+}
 
-  if (dayStatSnapshot !== null && dayStatSnapshot.timestamp.notEqual(dayStartUnixBigInt) ) {
-    let totalCollateral = dayStatSnapshot.totalCollateralAmount
-    dayStatSnapshot.id = dayStatSnapshot.timestamp.toString()
-    dayStatSnapshot.save()
-    
-    dayStatSnapshot = new DayStatSnapshot('last')
-    dayStatSnapshot.totalCollateralAmount  = totalCollateral
-    dayStatSnapshot.timestamp = dayStartUnixBigInt
+function getTransactionDetail(
+  vaultId: BigInt,
+  amount: BigInt,
+  vault: Vault,
+  timestamp: BigInt,
+  action: string,
+  transactionHash: string,
+  debtAmount: BigInt
+): VaultHistory {
+  const vaultHistory = new VaultHistory(transactionHash);
+  vaultHistory.totalEthCollateralAmount = vault.collateralAmount;
+  vaultHistory.action = action;
+  vaultHistory.vaultId = vaultId;
+  vaultHistory.timestamp = timestamp;
+  if (action === "MINT_SHORT" || action === "BURN_SHORT") {
+    vaultHistory.oSqthAmount = amount;
```

Comment:
Don't we need to add vaultHistory.ethCollateralAmount = amount for MINT_SHORT & BURN_SHORT tx types?

Reply:
The amount in the event params for MINT_SHORT & BURN_SHORT is the osqth minted & burnt respectively. No ethCollateral amount there.

Reply:
Yep that's right